src/puppetlabs/puppetdb/http/command.clj

```diff
@@ -21,21 +21,31 @@
 
 (def valid-commands-str (str/join ", " (sort (vals command-names))))
 
+(defn coerce-to-int
+  [s]
+  (if (number? s) s (Integer. s)))
+
 (defn validate-command-version
   [app]
-  (fn [{:keys [body-string] :as req}]
-    (let [{:keys [command version]} (json/parse-string body-string true)
+  (fn [{:keys [body-string params param-post?] :as req}]
+    (let [{:strs [command version]} (if param-post?
+                                      params
+                                      (json/parse-string body-string))
+          command (str/replace command "-" " ")
```

Comment:
I'd be on board with this change, but I feel like it's enough of a docs/code change for its own patch. It would break cross-version imports from 3->4 (which we don't guarantee) without some kind of replace call like this one -- don't know that we care.